pkg/sql/parser/sql.y

```diff
@@ -2314,6 +2329,61 @@ partitioned_backup_list:
     $$.val = append($1.partitionedBackups(), $3.partitionedBackup())
   }
 
+// Optional restore options.
+opt_with_restore_options:
+  WITH restore_options_list
+  {
+    $$.val = $2.restoreOptions()
+  }
+| WITH OPTIONS '(' restore_options_list ')'
+  {
+    $$.val = $4.restoreOptions()
+  }
+| /* EMPTY */
+  {
+    $$.val = &tree.RestoreOptions{}
+  }
+
+restore_options_list:
+  // Require at least one option
+  restore_options
+  {
+    $$.val = $1.restoreOptions()
+  }
+| restore_options_list ',' restore_options
+  {
+    if err := $1.restoreOptions().CombineWith($3.restoreOptions()); err != nil {
+      return setErr(sqllex, err)
+    }
+  }
+
+// List of valid restore options.
+restore_options:
+  ENCRYPTION_PASSPHRASE '=' string_or_placeholder
+  {
+    $$.val = &tree.RestoreOptions{EncryptionPassphrase: $3.expr()}
+  }
+| INTO_DB '=' database_name
```

Comment:
do you feel like this is an improvement over the current state of affairs? adding all of these extra keywords when we could just manipulate a map in go seems questionable.

Reply:
We went this way with BACKUP's option since it makes it more self-documenting -- the parser can actually point to invalid options, the diagram generator can correctly print the valid options, etc

Reply:
that makes sense, sounds good.